src/App.ts

```diff
@@ -719,7 +719,38 @@ export default class App {
       };
     };
 
-    // Set body and payload (this value will eventually conform to AnyMiddlewareArgs)
+    // Set body and payload
+    // TODO: this value should eventually conform to AnyMiddlewareArgs
+    let payload: any = {};
```

Comment:
I didn't spend much time trying to figure out the various possible types the `payload` variable could end up being, so I took the lazy way and set this as `any`. Sounds like it is related to the TODO comment above? Let me know, happy to dig in here further if the team thinks it is worthwhile.

Reply:
Trying to use a union type like `DialogSubmitAction | WorkflowStepEdit | SlackShortcut | KnownEventFromType<string> | ...`  may make this safer. If we go with `any` here, we will lose the type safety at the `listenerArgs` initialization below.

<img width="827" alt="Screen Shot 2021-08-11 at 09 18 19" src="https://user-images.githubusercontent.com/19658/128951071-aafce0e5-ab49-4f15-a1f0-60b248bdc1ef.png">